src/UIKit/UIView.cs

```diff
@@ -176,6 +176,15 @@ public static Task<bool> AnimateAsync (double duration, NSAction animation)
 		{
 			return AnimateNotifyAsync (duration, animation);
 		}
+
+		public UIImage Capture (bool afterUpdates = false)
+		{
+			UIGraphics.BeginImageContextWithOptions (Bounds.Size, Opaque, 0.0f);
+			DrawViewHierarchy (Bounds, afterUpdates);
+			var snapshot = UIGraphics.GetImageFromCurrentImageContext ();
+			UIGraphics.EndImageContext ();
```

Comment:
worth a `try`...`finally` to ensure things remain balanced (e.g. an overridden `DrawViewHierarchy` could throw a managed exception).